### PR TITLE
Added filter type allow and allowing name attribute

### PR DESCRIPTION
**Pull request by JonasKunze**

### CODE REVIEW

File: src/noit_check_rest.c [hunk 1]
```diff
@@ -396,7 +396,7 @@ rest_show_check(mtev_http_rest_closure_t *restc,
   xmlNodePtr node, root, attr, config, state, tmp, anode;
   uuid_t checkid;
   noit_check_t *check;
-  char xpath[1024], *uuid_conf, *module = NULL, *value = NULL;
+  char xpath[1024], *uuid_conf = NULL, *module = NULL, *value = NULL;
```
Comment (postwait) [inline src/noit_check_rest.c]: Line 425 returns without releasing doc and uuid_conf.

File: src/noit_filters_rest.c [hunk 2]
```diff
@@ -118,18 +118,26 @@ make_conf_path(char *path) {
...
-validate_filter_post(xmlDocPtr doc) {
+validate_filter_post(xmlDocPtr doc, char *name) {
   xmlNodePtr root, r;
+  char *old_name;
   root = xmlDocGetRootElement(doc);
   if(!root) return NULL;
   if(strcmp((char *)root->name, "filterset")) return NULL;
-  if(xmlHasProp(root, (xmlChar *)"name")) return NULL;
+
+  old_name = (char *)xmlGetProp(root, (xmlChar *)"name");
```
Comment (JonasKunze) [inline src/noit_filters_rest.c]: I do this in line 132 and 135

File: src/noit_check_rest.c (hunk 1 above)
Comment (JonasKunze) [inline src/noit_check_rest.c]: I guess you mean pobj instead of doc (see last commit)
doc is not initialized in line 425 yet

**Comment by JonasKunze** [comment]
I've split this into two PRs:
https://github.com/circonus-labs/reconnoiter/pull/322
https://github.com/circonus-labs/reconnoiter/pull/323

**Pull request closed without merging**